tests/test_helper.hpp

```diff
@@ -5,6 +5,10 @@
 #include <iomanip>
 #include <sycl/sycl.hpp>
 
+#undef __FAST_MATH__
```

Comment:
I think we should let the user control what they want to do.
If they want to use __fast_math__ let's do so, they will just need to adjust the tolerance if needed (maybe not as the host compiler may also use fast math)

Reply:
Note that the undefined are part of the `test_helper` header, not complex. So for the user, `fast_math` is still enabled when using it, apart from the test.

Reply:
My preference is to do this via compiler flags, rather than programmatically. The CI for example can just use in CMAKE_CXX_FLAGS -fp-model=precise. One possibility, if you are concerned about people running tests manually and forgetting to change compiler flags, would be to issue a warning if _SYCL_EXT_CPLX_FAST_MATH is set (after the header is included). Eventually, disabling some tests and adjusting tolerence so tests can pass in both modes would be ideal.